conda_build/index.py

```diff
@@ -5,78 +5,155 @@
 from __future__ import absolute_import, division, print_function
 
 import bz2
+from collections import defaultdict
 import contextlib
 from datetime import datetime
 from functools import partial
+from glob import glob
 import json
 import logging
 from numbers import Number
 import os
+from os.path import basename, dirname, getmtime, getsize, isdir, isfile, join, splitext
+from shutil import copy2
 import tarfile
-from os.path import isfile, join, getmtime, basename, getsize
 
 from jinja2 import Environment, PackageLoader
+import yaml
 
-from conda_build.utils import file_info, get_lock, try_acquire_locks
-from conda_build import utils, conda_interface
-from .conda_interface import PY3, md5_file, url_path, CondaHTTPError, get_index, human_bytes
+from . import conda_interface, utils
+from .conda_interface import (CondaHTTPError, VersionOrder, get_index, human_bytes, md5_file,
+                              url_path)
+from .utils import file_info, get_lock, rm_rf, try_acquire_locks
+
+log = logging.getLogger(__name__)
 
 local_index_timestamp = 0
 cached_index = None
 local_subdir = ""
 cached_channels = []
 
 
-def read_index_tar(tar_path, lock, locking=True, timeout=90):
-    """ Returns the index.json dict inside the given package tarball. """
-    locks = []
-    if locking:
-        locks = [lock]
-    with try_acquire_locks(locks, timeout):
-        with tarfile.open(tar_path) as t:
-            try:
-                return json.loads(t.extractfile('info/index.json').read().decode('utf-8'))
-            except EOFError:
-                raise RuntimeError("Could not extract %s. File probably corrupt."
-                    % tar_path)
-            except OSError as e:
-                raise RuntimeError("Could not extract %s (%s)" % (tar_path, e))
-            except tarfile.ReadError:
-                raise RuntimeError("Could not extract metadata from %s. "
-                                "File probably corrupt." % tar_path)
+def get_build_index(subdir, bldpkgs_dir, output_folder=None, clear_cache=False,
```

Comment:
This function is not modified in this PR, just shifted to a new location in the file.  No need to review.